### PR TITLE
Feature/whitelisting

### PR DESCRIPTION
**Pull request by brindy**
Reviewer: Mia
Asana: https://app.asana.com/0/0/427251562039378/f
CC:

**Description**:

Add support for whitelisting.

**Steps to test this PR**:

Test 1:
1. Visit a tracker heavy site
2. Note blocking taking place
3. Turn off blocking for this site
4. Use the app logs and/or javascript console to observe no blocking
5. Turn on blocking for this site
6. Note blocking taking place

Test 2:
1. Visit a tracker heavy site
2. Note blocking taking place
3. Turn off blocking for this site
4. Ideally by following a link on that site, navigate to another site
5. Note blocking taking place

Test 3:
1. Visit a tracker heavy site
2. Note blocking taking place
3. Visit settings and disable blocking
4. Return to website
5. Use the app logs and/or javascript console to observe no blocking
6. Visit settings and enable blocking
7. Return to website
8. Note blocking place

###### Reviewer Checklist:
- [ ] **Ensure the PR solves the problem**
- [ ] **Review every line of code**
- [ ] **Ensure the PR does no harm by testing the changes thoroughly**
- [ ] **Get help if you're uncomfortable with any of the above!**
- [ ] Determine if there are any quick wins that improve the implementation

###### PR DRI Checklist:
- [x] Get advice or leverage existing code
- [x] Agree on technical approach with reviewer (if the changes are nuanced)
- [x] Ensure that there is a testing strategy (and documented non-automated tests)
- [x] Ensure there is a documented monitoring strategy (if necessary)
- [x] Consider systems implications (Database connections, Grafana stats, CPU)

### CODE REVIEW

File: Core/blockerdata.js [hunk 1]
```diff
@@ -32,8 +32,35 @@ duckduckgo_decodeBase64 = function(s) {
...
     easylistPrivacy: {}
 
 }
+
+try {
+
+    ABPFilterParser.parse(function() {
+
+        var easylistData = duckduckgo_decodeBase64("${easylist_general}")
+        console.log("Easylist: " + easylistData.substring(0, 100))
```
Comment (subsymbolic) [inline Core/blockerdata.js]: We should probably switch this logging off when in release mode, happy to capture that in a separate task though.

File: Core/WKWebViewExtension.swift [hunk 2]
```diff
@@ -35,10 +35,16 @@ extension WKWebView {
         return webView
     }
     
-    public func loadScripts() {
+    public func loadScripts(contentBlocker: ContentBlockerConfigurationStore) {
+        let whitelist = contentBlocker.domainWhitelist.reduce("{", { (result, next) -> String in
+            let separator = result != "{" ? ", " : ""
+            return "\(result)\(separator) \"\(next)\" : true"
+        }).appending("}")
+        print("whitelist:", whitelist)
```
[the remark this answers is not in the document]
Comment (brindy) [reply Core/WKWebViewExtension.swift]: yep - was only meant to be there during dev, sorry

File: Core/blockerdata.js (hunk 1 above)
Comment (brindy) [reply Core/blockerdata.js]: can do - won't affect performance if the console is not open/attached though (having read the webkit code!)

Comment (subsymbolic) [reply Core/blockerdata.js]: My logging concerns are more around privacy.

Comment (subsymbolic) [reply Core/blockerdata.js]: Cool I've made a separate task in Asana https://app.asana.com/0/414709148257752/433059546511657

**Pull request merged**